### ISSUE TITLE
Move documentation dependencies into separate requirements 

### ISSUE THREAD

**Issue by verbosemode** [mpenning/ciscoconfparse#258]
It would be nice if documentation dependencies like Sphinx would not be installed by default. This would reduce the direct and transitive dependencies quite a bit. 

Those dependencies could be moved to requirements-docs.txt. For Poetry it should be possible to create a docs group, like for development dependencies.

The increase in dependencies has happened after 1.6.52.

An installation of 1.6.52 resulted in 5 new packages in a new virtual environment. Upgrading to 1.6.53 pulled in about 90 new packages.

I just wanted to raise the topic here first, but I can create a PR, if this proposal looks like a sensible solution to you.

**Comment by mpenning** [comment]
Your wish is...

🪄 done... see version 1.7.2.

**Comment by verbosemode** [comment]
Thank you for looking into this. Unfortunately the dependencies like sphinx, black, pylint, pre-commit are still installed in a fresh virtual environment. I have tested it with Poetry and then with pip only.

These packages are listed as dependencies in the latest [wheel](https://pypi.org/project/ciscoconfparse/#files) in `ciscoconfparse-1.7.3.dist-info/METADATA`.

Maybe Poetry packages it that way, because they are listed in the dependencies of `pyproject.toml`.

```
Requires-Dist: Sphinx (==4.3.0)
Requires-Dist: black (>=20.8b1)
Requires-Dist: deprecat (>=2.1.1)
Requires-Dist: dnspython (>=2.2.0,<3.0.0)
Requires-Dist: loguru (==0.6.0)
Requires-Dist: packaging (>21.0)
Requires-Dist: passlib (>=1.7.4,<2.0.0)
Requires-Dist: poetry (>=1.2.2)
Requires-Dist: poetry-core (>=1.3.2)
Requires-Dist: pre-commit (==2.20.0)
Requires-Dist: pylint (>=2.15.5)
Requires-Dist: pyroma (>=4.0)
Requires-Dist: pytest (>=7.1.3)
Requires-Dist: sphinx-bootstrap-theme (==0.8.0)
Requires-Dist: toml (>=0.10.2)
Requires-Dist: virtualenv (>=20.16.5,<20.17.0)
Requires-Dist: virtualenvwrapper (>=4.8.4,<4.9.0)
```

I have played around with Poetry's [dependency groups](https://python-poetry.org/docs/managing-dependencies/#dependency-groups), since I was not entirely familiar with them. Maybe they would suit your development workflow, since you're already using Poetry for project management. 

This is a modified version of the `pyproject.toml`, where the dependencies are split up in groups.

```
[tool.poetry.dependencies]
python = ">=3.7.2,<4.0.0"
passlib = "^1.7.4"
dnspython = "^2.2.0"
loguru = "0.6.0"
toml = ">=0.10.2"
deprecat = ">=2.1.1"

[tool.poetry.group.dev.dependencies]
pyroma = ">=4.0"
black = ">=20.8b1"
pre-commit = "2.20.0"
virtualenv = "~=20.16.5"
virtualenvwrapper = "~=4.8.4"
poetry = ">=1.2.2"
poetry-core = ">=1.3.2"
pylint = ">=2.15.5"
packaging = ">21.0"

[tool.poetry.group.test.dependencies]
pytest = ">=7.1.3"

[tool.poetry.group.docs.dependencies]
Sphinx = "4.3.0"
sphinx-bootstrap-theme = "0.8.0"
```

This also allows you to generate the requrements files via `poetry export` like this:

Contains only ciscoconfparse runtime dependencies

```poetry export --output=requirements.txt```

Separate files for dev/docs/test:

```
poetry export --only=dev --output=requirements-dev.txt
poetry export --only=docs --output=sphinx-doc/requirements.txt 
poetry export --only=test --output=requirements-test.txt
```

Combined file for dev/docs/test:

```poetry export --with=dev,docs,test --output=requirements-dev.txt```


The exact version pinnings might need to be tweaked a bit, since the Sphinx version in sphinx-doc/requirements.txt is not compatible with Python 3.7, but >= 3.8 for example.

**Comment by mpenning** [comment]
Thanks for your patience... I think it's finally fixed in 1.7.5... please verify, but it looked good to me in a fresh virtualenv.

**Comment by verbosemode** [comment]
Awesome, thanks for your tenacity  on this and for writing ciscoconfparse. My favorite tool for sifting through Rancid backups full of config files :-)

I've tested it today in a fresh venv and can confirm it is working.
